### DISCUSSION TITLE
stories_things/basic_content/inline/

### DISCUSSION THREAD

**Discussion by giscus** [Light-City/light-comments#17] [Announcements]
# stories_things/basic_content/inline/

Stories About C Plus Plus

https://light-city.github.io/stories_things/basic_content/inline/

**Comment by Austin-Jack** [comment]
``` 
Base *ptr = new Derived();
ptr->who();
```
这段代码上面的注释意思是编译器选择为非内联吗，如果是这样的话"不能为内联"这种说法让我以为inline对于虚函数会编译报错

**Reply by miaoxinxrct** [reply]
你显式告诉编译器inline的时候，编译器是要评估你的建议的，他会自己做一个优化，评估是否需要内联，所以对于编译器来说，你只是个建议者罢了，而编译器自身才是主导者
